src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/LocalHelpers.cs

```diff
@@ -192,15 +192,24 @@ public static string ExecuteCommand(string command, string arguments, ILogger lo
             return output;
         }
 
-        private static void ExecuteGitShallowSparseCommand(string gitLocation, string arguments, ILogger logger, string workingDirectory)
+        /// <summary>
+        ///     Execute a git command
+        /// </summary>
+        /// <param name="gitLocation">Location of git.exe</param>
+        /// <param name="arguments">Arguments to git</param>
+        /// <param name="logger">Logger</param>
+        /// <param name="workingDirectory">Working directory</param>
+        /// <param name="secretToMask">Mask this secret when calling the logger.</param>
+        private static void ExecuteGitCommand(string gitLocation, string arguments, ILogger logger, string workingDirectory, string secretToMask = null)
         {
-            using (logger.BeginScope("Executing command git {arguments} in {workingDirectory}...", arguments, workingDirectory))
+            string maskedArguments = arguments.Replace(secretToMask, "***");
```

Comment:
I think Replace throws if the first value is null or empty, so we might need a check